### ISSUE TITLE
ProgressBar's knobBefore will not disappear when at value of 0

### ISSUE THREAD

**Issue by BoBIsHere86** [libgdx/libgdx#6533]
#### Issue details
A progress bar's knobBefore will not disappear when the Value is set to 0. Even after having made sure the minWidth of the knobBefore drawable was 0. 
This issue seems to be related to c6b7119a53c94086ce04f07f62d2bf44f8cf0739
While Testing I made a subclass of ProgressBar and reverted the changes for Math.max(bgBottomHeight, position); back to Math.max(Math.min(0, bgBottomHeight), position);  and Math.max(bgLeftWidth, position); back to Math.max(Math.min(0, bgLeftWidth), position);  and it began to work as expected.

#### Reproduction steps/code
Made a simple barebones project. Using a skin with a just a drawable for the background and knobBefore.
ProgressBar min/max, 0 to 50, value set to 0;

#### Version of LibGDX and/or relevant dependencies
1.9.12, 1.9.13, 1.9.14 and 1.10.0
![progressbarrIssue](https://user-images.githubusercontent.com/55356015/117995771-97f18000-b30f-11eb-947c-3a2e09626720.png)

#### Please select the affected platforms
- [x] Android
- [x] iOS
- [x] HTML/GWT
- [x] Windows
- [x] Linux
- [x] MacOS

**Comment by obigu** [comment]
Duplicate? https://github.com/libgdx/libgdx/issues/6259

**Comment by BoBIsHere86** [comment]
> Duplicate? #6259 

I came across this issue in my search. But the workaround suggested, didn't have any effect. The only solution so far, was reverting that last change I mentioned above. While it might be related, I thought it was at least worth placing the issue. As it's been here since at least 1.9.12

**Comment by NathanSweet** [comment]
We first need to decide how before/after drawables should be shown. I guess they should be to the edge of the knob. Before:
![](http://n4te.com/x/1899-rBqS.png)
After:
![](http://n4te.com/x/1898-4wT2.png)
That allows knobs to have blank space on the left/right to let the before/after drawable show through.

**Comment by BoBIsHere86** [comment]
Thanks Nate!
Just a follow up, stating that I've tried this change out in the latest snapshot and it fixed the issue.

Also, I would like to just chime in and say that one of the reasons I brought this issue up was that I've noticed a lot of people are told to use the ProgressBar actor as a way to quickly make character health/energy bars. And the issue with the knobBefore/knobAfter would nullify that. I know it's not the progress bar's main purpose. It's just nice to have, though.
